x-pack/plugins/triggers_actions_ui/public/application/sections/rules_list/components/rules_list_auto_refresh.tsx

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useState, useRef } from 'react';
+import moment from 'moment';
+import { EuiFlexGroup, EuiFlexItem, EuiText, EuiAutoRefreshButton } from '@elastic/eui';
+
+interface RulesListAutoRefreshProps {
+  lastUpdate: string;
+  onRefresh: () => void;
+}
+
+export const RulesListAutoRefresh = (props: RulesListAutoRefreshProps) => {
+  const { lastUpdate, onRefresh } = props;
+
+  const [isPaused, setIsPaused] = useState<boolean>(false);
+  const [refreshInterval, setRefreshInterval] = useState<number>(5 * 60 * 1000);
+  const cachedOnRefresh = useRef<() => void>(() => {});
+  const timeout = useRef<number | undefined>(undefined);
+
+  useEffect(() => {
+    cachedOnRefresh.current = onRefresh;
```

Comment:
Nice trick here!

Reply:
yea trying to use `useCallback` with timers is super annoying because of stale closures haha